nixos/modules/services/hardware/amdvlk.nix

```
@@ -0,0 +1,61 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.hardware.amdgpu.amdvlk;
+in {
+  options.hardware.amdgpu.amdvlk = {
+    enable = lib.mkEnableOption "AMDVLK Vulkan driver";
+
+    package = lib.mkPackageOption pkgs "amdvlk" { };
+
+    supportExperimental.enable = lib.mkEnableOption "Experimental features support";
+
+    support32Bit.enable = lib.mkEnableOption "32-bit driver support";
+    support32Bit.package = lib.mkPackageOption pkgs.driversi686Linux "amdvlk" { };
+
+    settings = lib.mkOption {
+      type = lib.types.attrs;
```

```suggestion
      type = lib.types.attrsOf lib.types.int;
```

---

Sometimes, the values may contain a string, in particular for `GpuProfilerConfig.LogDirectory` it would be a directory path. See https://github.com/GPUOpen-Drivers/AMDVLK?tab=readme-ov-file#pal-gpuprofiler-layer

---

```suggestion
      type = with lib.types; attrsOf (either str int);
```

Yes! Please add a note in the 24.11 release notes :)

---

Done in https://github.com/NixOS/nixpkgs/pull/318175/commits/85814898f2105be9960a2e234e5f16ba8d8ac265

Added 24.11 release notes in https://github.com/NixOS/nixpkgs/pull/318175/commits/b2f1ce3c111364629444363dc325bfbf6ca2cb65